cmd/snap-update-ns/update_test.go

```
@@ -20,16 +20,302 @@
+	defer restore()
+
+	// Perform the update, this will modify assumptions.
+	c.Check(as.PastChanges(), HasLen, 0)
+	c.Assert(update.ExecuteMountProfileUpdate(ctx), IsNil)
+	c.Check(as.PastChanges(), HasLen, 1)
+	c.Check(as.PastChanges(), DeepEquals, []*update.Change{{
+		Action: update.Mount,
+		Entry:  entry,
+	}})
+}
+
+func (s *updateSuite) TestSyntheticChanges(c *C) {
+	// When a mount change is performed it may cause additional mount changes
+	// to be performed, that were needed internally. Such changes are recorded
+	// and saved into the current profile.
+	var saved *osutil.MountProfile
+	ctx := &testProfileUpdateContext{
+		loadDesiredProfile: func() (*osutil.MountProfile, error) {
+			return &osutil.MountProfile{Entries: []osutil.MountEntry{
+				{Dir: "/subdir/mount"},
+			}}, nil
+		},
+		saveCurrentProfile: func(profile *osutil.MountProfile) error {
+			saved = profile
+			return nil
+		},
+		neededChanges: func(old, new *osutil.MountProfile) []*update.Change {
+			return []*update.Change{
+				{Action: update.Mount, Entry: osutil.MountEntry{Dir: "/subdir/mount"}},
+			}
+		},
+		performChange: func(change *update.Change, as *update.Assumptions) ([]*update.Change, error) {
+			// If we are trying to mount /subdir/mount then synthesize a change
+			// for making a tmpfs on /subdir.
+			if change.Action == update.Mount && change.Entry.Dir == "/subdir/mount" {
+				return []*update.Change{
+					{Action: update.Mount, Entry: osutil.MountEntry{Dir: "/subdir", Type: "tmpfs"}},
+				}, nil
+			}
+			return nil, nil
+		},
+	}
+	restore := ctx.MockRelatedFunctions()
+	defer restore()
+	c.Assert(update.ExecuteMountProfileUpdate(ctx), IsNil)
+	c.Check(saved, DeepEquals, &osutil.MountProfile{Entries: []osutil.MountEntry{
+		{Dir: "/subdir", Type: "tmpfs"},
+		{Dir: "/subdir/mount"},
+	}})
+}
+
+func (s *updateSuite) TestCannotPerformContentInterfaceChange(c *C) {
+	// When performing a mount change for a content interface fails we simply
```

`...fails, we simply ignore the error and carry on ..`